pkg/graph/builder.go

```diff
@@ -292,14 +292,24 @@ func (b *Builder) buildRGResource(rgResource *v1alpha1.Resource, namespacedResou
 			return nil, fmt.Errorf("failed, CEL expressions are not supported for CRDs, resource %s", rgResource.ID)
 		}
 	} else {
-
 		// 4. Emulate the resource, this is later used to verify the validity of the
 		//    CEL expressions.
 		emulatedResource, err = b.resourceEmulator.GenerateDummyCR(gvk, resourceSchema)
 		if err != nil {
 			return nil, fmt.Errorf("failed to generate dummy CR for resource %s: %w", rgResource.ID, err)
 		}
 
+		// Add metadata fields to emulated resource
+		if emulatedResource.Object["metadata"] == nil {
+			emulatedResource.Object["metadata"] = map[string]interface{}{
+				"name": "dummy-name",
+				"creationTimestamp": "2024-01-01T00:00:00Z",
+				"namespace": "default",
+				"labels": map[string]interface{}{},
+				"annotations": map[string]interface{}{},
+			}
+		}
```

Comment:
I believe that we already set fake metadata on emulated resources here https://github.com/kro-run/kro/blob/main/pkg/graph/emulator/emulator.go#L70-L74